filewatcher/filewatcher.go

```diff
@@ -210,3 +230,45 @@ func New(ctx context.Context, cfg Config) (*Result, error) {
 
 	return res, nil
 }
+
+// NewMockFilewatcher returns a pointer to a new MockFileWatcher object
+// initialized with the given io.Reader and Parser.
+func NewMockFilewatcher(r io.Reader, parser Parser) (*MockFileWatcher, error) {
```

Comment:
Any reason why this should be in `filewatcher` rather than `filewatcher_test`?

Reply:
if it's in `filewatcher_test` it can only be used by tests under `filewatcher`, the intention here is to make it usable by other packages using filewatcher.

Reply:
That makes sense. Which package would also use it?

Reply:
currently `experiments`, but any package using filewatcher could benefit from it.

Reply:
`experiments` is switching to use the interface, is this for testing purposes only? It seems the interface is added only to allow for supplying a mock implementation in tests but not another implementation for non-test code.

Reply:
Oh sorry, I meant `secrets`, not `experiments`.

Reply:
I can't find it anywhere besides tests, speaking of `NewMockFilewatcher` here.

Reply:
it's in `secrets/testing.go`.

Reply:
It's not going to be used anywhere besides tests likely, but we can't put it in `filewatcher_tests` because `_test` packages aren't usable externally.  The reason for including these things is to make it easier to write tests for code involving file watchers.

Right now, only `secrets` uses it, but `experiments` could too as well as any custom code using file watchers in services.